bin/build-service-images

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+set -e
+
+SELF=$(basename $0)
+
+function usage
+{
+  echo "Usage: $SELF [-p|--parallel -h] [service_name ...]"
+  echo "Builds the images for all listed service names that have matching sub-directories in the 'services' directory."
+  echo "Builds all images if none are listed."
+  echo ""
+  echo "-p - run builds in parallel"
+  echo "-h - print this message"
+}
+
+PARALLEL=0
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+  -p|--parallel)
+  PARALLEL=1
+  shift
+  ;;
+  -h|--help)
+  usage
+  exit 0
+  ;;
+  *)
+  break
+  ;;
+esac
+done
+
+# build all the image in the 'services' directory
+function build_all {
+  parallel=$1
+  for dir in services/*; do
+    pushd "$dir"
+    if [[ $parallel -eq 1 ]]; then
+      ./bin/build-image &
+    else
+      ./bin/build-image
+    fi
+    popd
+  done
+}
+
+if [[ "$#" -eq 0 ]]; then
```

Comment:
We should do the PID tracking thing to capture errors anytime we do things in parallel.

Reply:
Done